mlir/include/mlir/Dialect/LLVMIR/LLVMAttrDefs.td

```diff
@@ -1213,6 +1213,17 @@ def LLVM_UndefAttr : LLVM_Attr<"Undef", "undef">;
 /// Folded into from LLVM::PoisonOp.
 def LLVM_PoisonAttr : LLVM_Attr<"Poison", "poison">;
 
+//===----------------------------------------------------------------------===//
+// DSOLocalEquivalentAttr
+//===----------------------------------------------------------------------===//
+
+/// Folded into from LLVM::DSOLocalEquivalentOp.
+def LLVM_DSOLocalEquivalentAttr : LLVM_Attr<"DSOLocalEquivalent",
+                                            "dso_local_equivalent"> {
```

Comment:
Can you try if you can derive from FlatSymbolRefAttr using something like this:

```
def LLVM_DSOLocalEquivalentAttr : LLVM_Attr<"DSOLocalEquivalent",
                                            "dso_local_equivalent", "FlatSymbolRefAttr"> {}
```
After all this is just a FlatSymbolRefAttr so inheritance would be nice.

Reply:
I tried this - it builds fine but the problem is in the `::get` methods, `DSOLocalEquivalentAttr` won't get something resembling the base class getters, so we end up only really able to do `DSOLocalEquivalentAttr::get(getContext())`, which doesn't setup any symbol. I also didn't find a clean way to set up the symbol after such instantiation.

Alternatively, if we add an extra param for the name, then what's the point of having the inheritance, it will store two strings (with the `FlatSymbolRefAttr` one empty).

I'm gonna move forward with passing a `FlatSymbolRefAttr` as param, if you can show me an example on how to use inheritance in a clean way I'm happy to tackle that in other PRs.

Reply:
Ok that would have to be implemented similar to FlatSymbolRefAttr then (in code https://github.com/llvm/llvm-project/blob/052a4b54a75364f13ea5001bba00691da4d8c239/mlir/include/mlir/IR/BuiltinAttributes.h#L869). Let's not do that then.